### PR TITLE
feat(js): Com 145 introduce novunextjs

### PR DESCRIPTION
**Pull request by BiswaViraj**
### What changed? Why was the change needed?

- created the `@novu/nextjs` package
- added next router as a default router push
### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: packages/nextjs/README.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
+# Novu's React SDK for building custom inbox notification experiences.
```
Comment (SokratisVidros) [inline packages/nextjs/README.md]:
```suggestion
# Novu's NextKS SDK for <Inbox />.
```

File: packages/nextjs/README.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+Novu provides the `@novu/react` a React library that helps to add a fully functioning Inbox to your web application in minutes. Let's do a quick recap on how you can easily use it in your application.
```
Comment (SokratisVidros) [inline packages/nextjs/README.md]:
```suggestion
Novu provides the `@novu/nextjs` library that helps to add a fully functioning <Inbox /> to your web application in minutes.
```

File: packages/nextjs/README.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+See full documentation [here](https://docs.novu.co/inbox/react/get-started).
+
+## Installation
+
+- Install `@novu/react` npm package in your react app
```
Comment (SokratisVidros) [inline packages/nextjs/README.md]:
```suggestion
- Install `@novu/nextjs` npm package in your react app
```

File: packages/nextjs/src/components/Inbox.tsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+'use client';
+
+import React from 'react';
+import { InboxProps, Inbox as RInbox } from '@novu/react';
+import { useRouter } from 'next/navigation';
+
+export const Inbox = React.memo((props: InboxProps) => {
+  const router = useRouter();
+
+  return <RInbox routerPush={router.push} {...props} />;
```
Comment (SokratisVidros) [inline packages/nextjs/src/components/Inbox.tsx:10]: ❤️

File: packages/nextjs/src/components/index.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
+'use client';
+
+import {
```
Comment (SokratisVidros) [inline packages/nextjs/src/components/index.ts:3]: Can't we export from @novu/react in a single file?

**Review by SokratisVidros — CHANGES_REQUESTED** [review]

File: packages/nextjs/src/components/index.ts (hunk 5 above)
Comment (BiswaViraj) [reply packages/nextjs/src/components/index.ts:3]: The problem is we need to export everything from `@novu/react` except the `Inbox`
I tried few methods like 
```
import * as NovuReact from'@novu/react
const {Inbox, ...rest} = NovuReact;
export {rest}
```
and other similar ways, they didn't worked :/

File: packages/nextjs/README.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+      options={{
+        subscriberId: 'SUBSCRIBER_ID',
+        applicationIdentifier: 'APPLICATION_IDENTIFIER',
+      }}
+    />
+  );
+}
+```
+
+## Use your own backend and socket URL
```
Comment (SokratisVidros) [inline packages/nextjs/README.md]: Please move this to the last example. The others are more important.

**Review by SokratisVidros — APPROVED** [review]

**Pull request merged**